src/components/order/info/Info.js

```diff
@@ -133,7 +133,9 @@ class Info extends React.PureComponent {
         const cart = this.state.cart;
         if (Number(index) !== this.state.selectedCollectionTypeIndex) {
             cart.delivery = undefined;
-            cart.pickup = undefined;
```

Comment:
this may break in the following case
1. choose delivery
2. fill details
3. switch to pickup
4. checkout cart without filling pickup details

Reply:
Checked.
System is not allowing to checkout the cart without filling pickup details.